src/Google/Ads/GoogleAds/Lib/V6/GoogleAdsLoggingServerStreamingCall.php

```diff
@@ -44,15 +45,44 @@ public function __construct(
         parent::__construct($innerCall);
         $this->lastRequestData = $lastRequestData;
         $this->googleAdsCallLogger = $googleAdsCallLogger;
+        $this->savedResponses = [];
     }
 
     /**
      * {@inheritdoc}
      */
     public function getStatus()
     {
-        $response = null;
         $status = parent::getStatus();
+        if (empty($this->savedResponses)) {
+            $this->logSummaryAndDetails($status);
+        } else {
+            foreach ($this->savedResponses as $response) {
+                $this->logSummaryAndDetails($status, $response);
+            }
+        }
+        return $status;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function responses()
+    {
+        foreach ($this->innerCall->responses() as $response) {
+            $this->savedResponses[] = $response;
+            yield $response;
+        }
+    }
+
+    /**
+     * Logs summary and the details of the given status and response.
+     *
+     * @param object|null $status the status to be logged
+     * @param object|null $response the response to be logged
+     */
+    private function logSummaryAndDetails(object $status, ?object $response = null)
```

Comment:
Should not `$status` be of type `?object` instead of `object`? I am asking because you use `object|null` in the comment block.